stackslib/src/net/api/posttransaction.rs

```diff
@@ -68,13 +70,18 @@ impl RPCPostTransactionRequestHandler {
 
     /// Decode a bare transaction from the body
     fn parse_posttransaction_octets(mut body: &[u8]) -> Result<StacksTransaction, Error> {
-        let tx = StacksTransaction::consensus_deserialize(&mut body).map_err(|e| {
-            if let CodecError::DeserializeError(msg) = e {
-                Error::DecodeError(format!("Failed to deserialize posted transaction: {}", msg))
-            } else {
-                e.into()
-            }
-        })?;
+        let tx =
+            StacksTransaction::consensus_deserialize_with_epoch(&mut body, StacksEpochId::latest())
```

Comment:
Revert -- we want to accept all syntactically valid transactions into the mempool since they may become semantically valid at an upcoming epoch boundary